massa-bootstrap/src/white_black_list.rs

```diff
@@ -35,14 +35,95 @@ impl SharedWhiteBlackList<'_> {
         })
     }
 
+    /// get the white list
+    pub fn get_white_list(&self) -> Option<HashSet<IpAddr>> {
+        self.inner.read().white_list.clone()
+    }
+
+    /// get the black list
+    pub fn get_black_list(&self) -> Option<HashSet<IpAddr>> {
+        self.inner.read().black_list.clone()
+    }
+
+    /// Add IP address to the black list
+    pub fn add_ips_to_blacklist(&self, ips: Vec<IpAddr>) -> Result<(), BootstrapError> {
+        let mut write_lock = self.inner.write();
+        let list = if let Some(black_list) = &mut write_lock.black_list {
+            black_list.extend(ips);
+            black_list.clone()
+        } else {
+            let hash_set = HashSet::from_iter(ips);
+            write_lock.black_list = Some(hash_set.clone());
+            hash_set
+        };
+        self.write_list_to_file(&self.black_path, &list)?;
+        Ok(())
+    }
+
+    /// Remove IPs address from the black list
+    pub fn remove_ips_from_blacklist(&self, ips: Vec<IpAddr>) -> Result<(), BootstrapError> {
+        let mut write_lock = self.inner.write();
+        if let Some(black_list) = &mut write_lock.black_list {
+            for ip in ips {
```

Comment:
@modship need your help here

Reply:
yes, i could use `difference` but that would require two iterations.
The first iteration by `difference` to find and return the result,
The second made by me with the result to remove elements.

(there are also instructions executed to transform the vector into a hashet).
Since the code is easily understandable here, i don"t think it's worth using `.difference()`